src/app/containers/App/index.jsx

```diff
@@ -9,7 +9,7 @@ import App from './App';
 
 export const ClientApp = props => (
   <BrowserRouter {...props}>
-    <App initialData={props.data} routes={props.routes} />
+    <App initialData={props.data} routes={props.routes} headers={props.data.headers} />
```

Comment:
The browser/client App gets its headers from `props.data.headers`, rather than `props.headers`, because the headers are persisted in `window.SIMORGH_DATA` for the client App, unlike the server App.

Reply:
what will happen here is headers are not defined on props.data?